src/main/scala/io/iohk/ethereum/faucet/FaucetHandler.scala

```diff
@@ -0,0 +1,99 @@
+package io.iohk.ethereum.faucet
+
+import akka.actor.{Actor, ActorLogging, Props}
+import akka.util.ByteString
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.faucet.FaucetStatus.WalletAvailable
+import io.iohk.ethereum.faucet.jsonrpc.WalletService
+import io.iohk.ethereum.keystore.Wallet
+import monix.execution.Scheduler.Implicits.global
+
+class FaucetHandler(walletService: WalletService, config: FaucetConfig) extends Actor with ActorLogging {
+
+  import FaucetHandler.FaucetHandlerMsg._
+  import FaucetHandler.FaucetHandlerResponse._
+
+  var wallet: Wallet = _
+
+  override def preStart(): Unit = {
+    log info "||=== Faucet Handler Hook ===||"
+    self ! Initialization
+  }
+
+  override def receive: Receive = unavailable()
+
+  private def unavailable(): Receive = {
+    case Status =>
+      sender() ! StatusResponse(FaucetStatus.FaucetUnavailable)
+
+    case Initialization => {
+      log.info("Initialization called (faucet unavailable)")
+      walletService.getWallet.runSyncUnsafe() match {
+        case Left(error) =>
+          log.error(s"Couldn't initialize wallet - error: $error")
+        //context become unavailable
```

Comment:
I think we should kill the faucet or try to recover

Reply:
This one is still not addressed. We could end up with the "running" but nonworking faucet. The better idea is to kill the whole app. It will be easier to notice (ex. in metrics)

Reply:
I like this idea.  So,  I separated 3 cases.
1_ if it can't load the wallet, the app shutdown.
2_ While trying to load the wallet with the configuration. At this moment the wallet is disable “FaucetUnavailable”.
3_ When the wallet is loaded, the status is WalletAvailable and we can use the faucet with normality.

If this approach results ok, I will add this description in the documentation.
What do you think?


I have pending another task of improvements, unit test in the supervisor actor.

Reply:
I think it is a good direction